libc/src/__support/arg_list.h

```diff
@@ -54,7 +54,8 @@ class MockArgList {
   }
 
   template <class T> LIBC_INLINE T next_var() {
-    ++arg_counter;
+    arg_counter =
```

Comment:
I now use the `MockArgList` to tell determine how big the "struct" needs to be to contain the arguments.

Reply:
in that case it'd be better to create a new `ArgList` that is intended to count sizes instead of just modifying the one for testing.

Reply:
I don't think anything directly depends on its output, but also the comments on it specifically say that it's for testing. Given how small `ArgList` is I'd rather have another copy than run into problems because a patch in the future didn't expect the testing version to be used in real code.